### PR TITLE
Implements registerable commands and custom require wrapper for local files

### PR DESCRIPTION
**Pull request by adigitalmonk**
Proposed set up for how we can register commands and load them.  This also has a thing for a global set up where we can save some things that are acceptable to have access to globally (like the config).

### CODE REVIEW

File: register.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+registerCommand('syn', {
```
Comment (dmasterson) [inline register.js]: Is this just illustrating the ability to register a one off? Combine in the `registerCommands`?

File: register.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    'src' : 'src/commands/syn.js',
+    'limit' : 20
+});
+
+registerCommands({
```
Comment (dmasterson) [inline register.js]: `showPermissions` is the only pluralized command of these. Fun fact, if the filename is incorrect here, it kills the bot completely.

File: src/commands/grantpermission.js [hunk 3]
```diff
@@ -1,7 +1,6 @@
 const join = require('path').join;
-const Command = require(join('..','command.js'));
-const permissions = require(join('..','permissions.js'));
-const Logger = require(join('..','logger.js'));
+const Command = grab('src/command.js');
+const Logger = grab('src/logger.js');
```
Comment (dmasterson) [inline src/commands/grantpermission.js]: If we are now utilizing a global space, shouldn't Logger be part of it?

File: src/permissions.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
-const config = require('./configuration.js');
 const fs = require('fs');
 const EventEmitter = require('events').EventEmitter;
 const path = require('path');
-const bootstrap = require('./bootstrap.js');
-const permissions_file = path.join(__dirname,'..','conf', config.getSetting('permissions_file'));
-const Logger = require('./logger.js');
+
+const permissions_file = path.join(__dirname, '..', 'conf', Settings.getSetting('permissions_file'));;
```
Comment (dmasterson) [inline src/permissions.js]: `;`-- please

Comment (adigitalmonk) [inline src/permissions.js]: This is why we can't have nice things.

File: register.js (hunk 2 above)
Comment (adigitalmonk) [inline register.js]: Oy, your command names.

File: register.js (hunk 1 above)
Comment (adigitalmonk) [inline register.js]: It was just illustrating, could easily combine.

Comment (adigitalmonk) [inline register.js]: I intend to improve upon the way the things are registered, but it'll suffice for now.  I created the Registrar to for this reason.

File: register.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    },
+    'revokepermission' : {
+        'src' : 'src/commands/revokepermission.js',
+        'limit' : 20
+    },
+    'showpermissions' : {
+        'src' : 'src/commands/showpermissions.js',
+        'limit' : 20
+    }
+});
```
Comment (dmasterson) [inline register.js]: Sure would be swell if you included shutdown too.

Comment (adigitalmonk) [inline register.js]: Added, but I didn't include the limit since it's not implemented yet anyways

File: global.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+global.Logger = new Logger();
+
+// Command registrar
+const Registrar = grab('src/registrar.js');
+let r = new Registrar();
+
+// Bind the registrar to the global commands
+global.registerCommand = r.registerCommand.bind(r);
+global.registerCommands = r.registerCommands.bind(r);
+global.getCommand = r.getCommand.bind(r);
```
Comment (dmasterson) [inline global.js]: Whitespace

File: global.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+global.registerCommands = r.registerCommands.bind(r);
+global.getCommand = r.getCommand.bind(r); 
+
+// Create the settings location
+global.Settings = undefined;
+
+// Create the permissions location
+global.Permissions = undefined;
+
+// Export nothing since we are just
```
Comment (dmasterson) [inline global.js]: Whitespace

**Comment by dmasterson** [comment]
Line 20 in interface also has whitespace issues. The `'line',` line.

**Comment by adigitalmonk** [comment]
Did whitespace kick your dog?

**Pull request merged**